src/NuGet.LibraryModel/LibraryIdentity.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Versioning;
+
+namespace NuGet.LibraryModel
+{
+    /// <summary>
+    /// Represents an entity that can be referenced by a Project, a NuGet Package, or another Project
+    /// </summary>
+    public class LibraryIdentity : IEquatable<LibraryIdentity>
+    {
+        public string Name { get; }
+
+        public string Type { get; }
+
+        public NuGetVersion Version { get; }
+
+        public LibraryIdentity(string name, string type, NuGetVersion version)
+        {
+            Name = name;
+            Type = type;
+            Version = version;
+        }
+
+        public override string ToString()
+        {
+            var name = Type + "/" + Name;
+            return name + " " + Version?.ToString();
+        }
+
+        public bool Equals(LibraryIdentity other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
```

Comment:
Don't know if there is a better way to do this, but these if checks can run twice if the Equals(object) calls this